apps/quran/tests/internal/test_hierarchy_workflow.py

```
@@ -0,0 +1,88 @@
+        self.assertEqual(words_res.status_code, 200, words_res.content)
+        words = words_res.json()
+        self.assertEqual(len(words), ayah_with_words["words_count"])
+        self.assertEqual([w["position_in_ayah"] for w in words], [1, 2])
+        self.assertEqual([w["text"] for w in words], ["بِسْمِ", "اللَّهِ"])
+
+    def test_hierarchy_drill_down_workflow_where_ayah_missing_should_return_ayah_not_found(self):
+        # Arrange
+        user = baker.make(User, email="workflow@example.com", is_active=True)
+        self.authenticate_user(user)
+
+        # Step 1–2: successful expands
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)